pkg/controller/kms/key/setup.go

```diff
@@ -0,0 +1,330 @@
+		cr.SetConditions(xpv1.Deleting())
+	case string(svcapitypes.KeyState_PendingImport):
+		cr.SetConditions(xpv1.Unavailable())
+	case string(svcapitypes.KeyState_Unavailable):
+		cr.SetConditions(xpv1.Unavailable())
+	}
+
+	// special case: if key is scheduled for deletion, abort early and do not schedule for deletion again
+	if obj.KeyMetadata.DeletionDate != nil {
+		return managed.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	return obs, nil
+}
+
+func preCreate(_ context.Context, cr *svcapitypes.Key, obj *svcsdk.CreateKeyInput) error {
+	return nil
+}
+
+func postCreate(_ context.Context, cr *svcapitypes.Key, obj *svcsdk.CreateKeyOutput, creation managed.ExternalCreation, err error) (managed.ExternalCreation, error) {
+	if err != nil {
+		return creation, err
+	}
+	meta.SetExternalName(cr, *obj.KeyMetadata.KeyId)
+
+	return managed.ExternalCreation{ExternalNameAssigned: true}, nil
+}
+
+type updater struct {
+	client svcsdkapi.KMSAPI
+}
+
+func (u *updater) update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*svcapitypes.Key)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	if cr.Spec.ForProvider.Description != nil {
+		if _, err := u.client.UpdateKeyDescriptionWithContext(ctx, &svcsdk.UpdateKeyDescriptionInput{
+			KeyId:       aws.String(meta.GetExternalName(cr)),
+			Description: cr.Spec.ForProvider.Description,
+		}); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errUpdate)
+		}
+	}
+
+	// Policy
+	if _, err := u.client.PutKeyPolicyWithContext(ctx, &svcsdk.PutKeyPolicyInput{
+		KeyId:      aws.String(meta.GetExternalName(cr)),
+		PolicyName: aws.String("default"),
+		Policy:     cr.Spec.ForProvider.Policy,
+	}); err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errUpdate)
+	}
+
+	// Tags
+	tagsOutput, err := u.client.ListResourceTagsWithContext(ctx, &svcsdk.ListResourceTagsInput{
+		KeyId: aws.String(meta.GetExternalName(cr)),
+	})
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errUpdate)
+	}
+	keepTags, removeTags := diffTags(cr.Spec.ForProvider.Tags, tagsOutput.Tags)
+	tags := []*svcsdk.Tag{}
+	for _, t := range keepTags {
+		tags = append(tags, &svcsdk.Tag{
+			TagKey:   t.TagKey,
+			TagValue: t.TagValue,
+		})
+	}
+	if _, err := u.client.TagResourceWithContext(ctx, &svcsdk.TagResourceInput{
+		KeyId: aws.String(meta.GetExternalName(cr)),
+		Tags:  tags,
+	}); err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+	untagKeys := []*string{}
+	for _, t := range removeTags {
+		untagKeys = append(untagKeys, t.TagKey)
+	}
+	if _, err := u.client.UntagResourceWithContext(ctx, &svcsdk.UntagResourceInput{
+		KeyId:   aws.String(meta.GetExternalName(cr)),
+		TagKeys: untagKeys,
+	}); err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+
+	// Enable / Disable
+	if cr.Spec.ForProvider.Enabled != nil && *cr.Spec.ForProvider.Enabled != *cr.Status.AtProvider.Enabled {
```

Comment:
We can get the values of bool pointers with `aws.BoolValue` to be safe.

Reply:
I think we can have this if check in `enableDisableKey` since there is already some conditioning happening there.